python/cuml/fil/fil.pyx

```diff
@@ -245,10 +246,56 @@ cdef class ForestInference_impl():
 
 class ForestInference(Base):
     """
+    ForestInference provides GPU-accelerated inference (prediction)
+    for random forest and boosted decision tree models.
+
+    This module does not support training models. Rather, users should
+    train a model in another package and save it in a
+    treelite-compatible format. (See https://github.com/dmlc/treelite)
+    Currently, LightGBM and XGBoost GBDT and random forest models are
+    supported.
+
+    Users typically create a ForestInference object by loading a
+    saved model file with ForestInference.load. The resulting object
+    provides a `predict` method for carrying out inference.
+
+    **Known limitations**:
+     * Trees are represented in a dense format, so a tree of depth k
+       will be stored in (2**k) - 1 nodes. This will be less space-efficient
+       for sparse trees.
+     * While treelite supports additional formats, only XGBoost and LightGBM
+       are tested
+     * LightGBM categorical features are not supported
+     * Inference uses a dense matrix format, which is efficient for many
```

Comment:
I'm trying to understand how this is different from the first bullet point. Does it mean both the serialized format AND the inputs to the `predict()` function are dense?

Reply:
Yes, dense trees is about the structure of the tree and dense data is about the structure of the 
X matrix. Maybe I can refer to the trees as "complete" rather than dense (everyone internally calls them dense though)